celery/backends/base.py

```diff
@@ -351,6 +351,54 @@ def encode_result(self, result, state):
     def is_cached(self, task_id):
         return task_id in self._cache
 
+    def _get_result_meta(self, result,
```

Comment:
particularly this block and related changes tests

Reply:
@auvipy Do you mean writing new tests?

Reply:
yes some more test